### ISSUE TITLE
eksctl hangs after waiting for CloudFormation stack even after CloudFormation stack is done.

### ISSUE THREAD

**Issue by hq6** [weaveworks/eksctl#4913]
Here's a cluster config. Note that we are using a VPC that has no outbound internet access.

```
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: hq6-test
  region: us-west-2
iam:
  withOIDC: true
vpc:
  id: vpc-001337d887de56c6f
  securityGroup: sg-0b6972aef0ea1a6b9
  sharedNodeSecurityGroup: sg-0b6972aef0ea1a6b9
  clusterEndpoints:
    privateAccess: true
    publicAccess: false
  subnets:
    private:
      us-west-2a: { id: subnet-03863755416c0aff4 }
      us-west-2b: { id: subnet-0bc69edc807043984 }
      us-west-2c: { id: subnet-028fea12f8d82e1d8 }
nodeGroups:
  - name: hq6-test-ng
    desiredCapacity: 2
    instanceType: t3.medium
    disableIMDSv1: true
    privateNetworking: true
```

When I run the command `eksctl create cluster -f hq6.yaml`, I get the following output.  The expected output is either success or error.
```
2022-03-07 17:11:19 [ℹ]  eksctl version 0.85.0
2022-03-07 17:11:19 [ℹ]  using region us-west-2
2022-03-07 17:11:19 [!]  warning, having public access disallowed will subsequently interfere with some features of eksctl. This will require running subsequent eksctl (and Kubernetes) commands/API calls from within the VPC.  Running these in the VPC requires making updates to some AWS resources.  See: https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html for more details
2022-03-07 17:11:19 [!]  security group rules may be added by eksctl; see vpc.manageSharedNodeSecurityGroupRules to disable this behavior
2022-03-07 17:11:19 [✔]  using existing VPC (vpc-001337d887de56c6f) and subnets (private:map[us-west-2a:{subnet-03863755416c0aff4 us-west-2a 10.170.115.0/24} us-west-2b:{subnet-0bc69edc807043984 us-west-2b 10.170.114.0/24} us-west-2c:{subnet-028fea12f8d82e1d8 us-west-2c 10.170.116.0/24}] public:map[])
2022-03-07 17:11:19 [!]  custom VPC/subnets will be used; if resulting cluster doesn't function as expected, make sure to review the configuration of VPC/subnets
2022-03-07 17:11:20 [ℹ]  nodegroup "hq6-test-ng" will use "ami-01ae27c716d8b85f7" [AmazonLinux2/1.21]
2022-03-07 17:11:20 [ℹ]  using Kubernetes version 1.21
2022-03-07 17:11:20 [ℹ]  creating EKS cluster "hq6-test" in "us-west-2" region with un-managed nodes
2022-03-07 17:11:20 [ℹ]  1 nodegroup (hq6-test-ng) was included (based on the include/exclude rules)
2022-03-07 17:11:20 [ℹ]  will create a CloudFormation stack for cluster itself and 1 nodegroup stack(s)
2022-03-07 17:11:20 [ℹ]  will create a CloudFormation stack for cluster itself and 0 managed nodegroup stack(s)
2022-03-07 17:11:20 [ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=us-west-2 --cluster=hq6-test'
2022-03-07 17:11:20 [ℹ]  Kubernetes API endpoint access will use provided values {publicAccess=false, privateAccess=true} for cluster "hq6-test" in "us-west-2"
2022-03-07 17:11:20 [ℹ]  CloudWatch logging will not be enabled for cluster "hq6-test" in "us-west-2"
2022-03-07 17:11:20 [ℹ]  you can enable it with 'eksctl utils update-cluster-logging --enable-types={SPECIFY-YOUR-LOG-TYPES-HERE (e.g. all)} --region=us-west-2 --cluster=hq6-test'
2022-03-07 17:11:20 [ℹ]
2 sequential tasks: { create cluster control plane "hq6-test",
    2 sequential sub-tasks: {
        4 sequential sub-tasks: {
            wait for control plane to become ready,
            associate IAM OIDC provider,
            2 sequential sub-tasks: {
                create IAM role for serviceaccount "kube-system/aws-node",
                create serviceaccount "kube-system/aws-node",
            },
            restart daemonset "kube-system/aws-node",
        },
        create nodegroup "hq6-test-ng",
    }
}
2022-03-07 17:11:20 [ℹ]  building cluster stack "eksctl-hq6-test-cluster"
2022-03-07 17:11:20 [ℹ]  deploying stack "eksctl-hq6-test-cluster"
2022-03-07 17:11:50 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:12:20 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:13:21 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:14:21 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:15:21 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:16:21 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:17:22 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:18:22 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:19:22 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:20:22 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:21:23 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:22:23 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:23:23 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:24:23 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"
2022-03-07 17:25:24 [ℹ]  waiting for CloudFormation stack "eksctl-hq6-test-cluster"

# Note that there is no more output, even though the wall-time is well past 17:25.
```

Note that CloudFormation console shows the cluster is done building, but I do not see updates to `~/.kube/config`.
The expected behavior is that a cluster is successfully created, but I see no messages indicating that the cluster is finished creating.

**Comment by hq6** [comment]
Quick update. I now see the following errors on kubelet:
```
Mar  8 19:57:54 ip-10-170-115-60 kubelet: I0308 19:57:54.355288   32141 prober.go:116] "Probe failed" probeType="Readiness" pod="kube-system/aws-node-ts2t4" podUID=021ac190-261a-435b-aabf-09f66fd5dc85 containerName="aws-node" probeResult=failure output="{\"level\":\"info\",\"ts\":\"2022-03-08T19:57:54.350Z\",\"caller\":\"/usr/local/go/src/runtime/proc.go:225\",\"msg\":\"timeout: failed to connect service \\\":50051\\\" within 5s\"}\n"
Mar  8 19:57:54 ip-10-170-115-60 kubelet: I0308 19:57:54.465137   32141 cni.go:239] "Unable to update cni config" err="no networks found in /etc/cni/net.d"
Mar  8 19:57:54 ip-10-170-115-60 kubelet: E0308 19:57:54.814096   32141 kubelet.go:2214] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized"
```

**Comment by nlpfollower** [comment]
Hello, were you able to resolve this issue?

**Comment by hq6** [comment]
Yes. 
IIRC, I ended up manually editing a K8 object to add proxy information for my company while eksctl was in this state, but I do not remember the details.

**Comment by nlpfollower** [comment]
Thanks for the reply. I'll play around with the K8 object. 

For others reading, while I wasn't able to speedup the CloudFormation, I was able to split my cluster into two ClusterConfigs: a slow one creating vpcs, gateways, etc. run with a `--without-nodegroup` flag. And another fast ClusterConfig that only spawns the nodes. I've also found that terminating the EC2s with AWS CLI directly is significantly faster, and doesn't interfere with eksctl cleanup process.
